bot/admin/test/xray/src/main/kotlin/XrayService.kt

```diff
@@ -582,15 +584,75 @@ class XrayService(
             testsPlans: List<String>,
             labelTestPlansMap: Map<String, String> = emptyMap()
     ): XrayTest? {
+        val steps = defineTestSteps(dialog)
+
+        val labels = linkedJira?.let { XrayClient.getLabels(it) } ?: emptyList()
+
+        val test = JiraTest(
+                jiraProject,
+                testName.invoke(labels),
+                "",
+                listOf("automatisation"),
+                testTypeField,
+                manualStepsField,
+                "cucumber-tock"
+        )
+        val jira = XrayClient.createTest(test)
+        if (linkedJira != null) {
+            XrayClient.linkTest(jira.key, linkedJira)
+            testsPlans.forEach {
+                XrayClient.addTestToTestPlan(jira.key, it)
+            }
+            if (labelTestPlansMap.isNotEmpty()) {
+                labels
+                        .filter { labelTestPlansMap.containsKey(it) }
+                        .forEach {
+                            XrayClient.addTestToTestPlan(jira.key, labelTestPlansMap[it]!!)
+                        }
+            }
+        }
+
+        steps?.forEach {
+            XrayClient.saveStep(jira.key, it)
+        }
+
+        XrayPrecondition.getPreconditionForUserInterface(dialog.actions.first().userInterfaceType)
+                ?.apply {
+                    XrayClient.addPrecondition(this, jira.key)
+                }
+
+        return XrayTest(jira.key)
+    }
+
+    fun updateXrayTest(dialog: DialogReport, testKey: String): XrayTest? {
+        val steps = defineTestSteps(dialog)
+
+        if (XrayClient.getIssue(testKey) == null) {
+            return null
+        }
+
+        // remove all existing test steps
+        val stepList = XrayClient.getTestSteps(testKey)
+        stepList.forEach { currentStep ->
+            XrayClient.deleteStep(testKey, currentStep.id.toInt())
```

Comment:
What about remaining tests if one deletion fails ? Do we assume others will fail too ?

Reply:
If steps deletion fails, en error would be raised and test update will be called again by testers. So, updateXrayTest() will be called again and it will try to delete steps.
No error handler will be added here.